### PR TITLE
Reduce build concurency mvnd.minThreads in CI builds

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- to fix mvnd failure in `Install` step of `Sytle Check` job in (https://github.com/apache/kyuubi/actions/runs/4403008223/jobs/7711886893#step:7:1012)
- by lowering `mvnd.minThreads` to reduce memory pressure in concurrent builds

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: build/mvnd [hunk 1]
```diff
@@ -25,7 +25,7 @@ _CALLING_DIR="$(pwd)"
 _COMPILE_JVM_OPTS="-Xms2g -Xmx2g -XX:ReservedCodeCacheSize=1g -Xss128m"
 
 if [ "$CI" ]; then
-  export MAVEN_CLI_OPTS="-Dmvnd.minThreads=8 --no-transfer-progress --errors --fail-fast -Dstyle.color=always"
+  export MAVEN_CLI_OPTS="-Dmvnd.minThreads=4 --no-transfer-progress --errors --fail-fast -Dstyle.color=always"
```
Comment (yaooqinn) [inline build/mvnd:28]: can we just use nproc to get the num of available processors

Comment (bowenliang123) [reply build/mvnd:28]: `--threads` supports getting by the num of available processors, but `mvnd.minThreads` does not. The config `mvnd.minThreads` is used on purpose as the fallback, and if `--threads` set later in command, the `mvnd.minThreads` will be ignored.

Comment (yaooqinn) [reply build/mvnd:28]: I mean

export MAVEN_CLI_OPTS="-Dmvnd.minThreads=$(nproc) --no-transfer-progress --errors --fail-fast -Dstyle.color=always"

Comment (bowenliang123) [reply build/mvnd:28]: Good suggestion. Will do it in a follow-up if necessary in future.
This PR mainly concerns lower the memory pressure on build concurrency. Let's see whether it helps this time.

**Pull request closed without merging**